migration/usage-metric-migration/src/main/java/io/camunda/migration/usagemetric/client/es/ElasticsearchUsageMetricMigrationClient.java

```diff
@@ -93,25 +106,31 @@ public <T> T findOne(
 
     try {
       final SearchResponse<T> response =
-          client.search(
-              b ->
-                  b.index(index)
-                      .from(0)
-                      .size(1)
-                      .query(getSearchQueryTransformer().apply(searchQuery)),
-              entityClass);
+          retryDecorator.decorate(
+              "Find one entity",
+              () ->
+                  client.search(
+                      b ->
+                          b.index(index)
+                              .from(0)
+                              .size(1)
+                              .query(getSearchQueryTransformer().apply(searchQuery)),
+                      entityClass),
+              res -> res.hits() == null || res.hits().hits().isEmpty());
 
       final var hits = response.hits().hits();
       return !hits.isEmpty() ? hits.getFirst().source() : null;
-    } catch (final IOException e) {
+    } catch (final Exception e) {
       throw new MigrationException(e);
     }
   }
 
   @Override
   public boolean getTask(final String taskId) throws MigrationException {
     try {
-      final var response = client.tasks().get(r -> r.taskId(taskId));
+      final GetTasksResponse response =
+          retryDecorator.decorate(
+              "Get reindex task", () -> client.tasks().get(r -> r.taskId(taskId)), res -> false);
```

Comment:
💭 I used `false` as the `retryPredicate`. We have a scheduler to poll the getTask anyway, so it will be retried a couple of times.